src/main/java/org/terasology/genome/breed/DiscreteBreedingAlgorithm.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2014 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.genome.breed;
+
+import org.terasology.genome.breed.mutator.GeneMutator;
+import org.terasology.utilities.random.FastRandom;
+
+import java.util.Arrays;
+
+/**
+ * A breeding algorithm that produces a diploid cross for Discrete traits with a Punnet Square.
+ */
+public class DiscreteBreedingAlgorithm implements BreedingAlgorithm {
+    private float mutationChance;
+    private GeneMutator geneMutator;
+
+    public DiscreteBreedingAlgorithm(float mutationChance, GeneMutator geneMutator) {
+        this.mutationChance = mutationChance;
+        this.geneMutator = geneMutator;
+    }
+
+    /**
+     * Check whether two organisms with the given genes can breed.
+     *
+     * @param genes1 The genes of the first organism
+     * @param genes2 The genes of the second organism
+     * @return Whether the two organisms can breed
+     */
+    @Override
+    public boolean canCross(String genes1, String genes2) {
+        validateGenes(genes1, genes2);
+
+        return (genes1.compareToIgnoreCase(genes2) == 0 ? true : false);
```

Comment:
This can be simplified to a single boolean operation. 
`return !(genes1.compareToIgnoreCase(genes2));`

Reply:
fixed, had done this because implicit type casting of int to boolean was returning an error